### PR TITLE
Basic frontend simulation

### PR DESCRIPTION
**Pull request by Lev-Stambler**
This PR relies on #82 and is half of the PR to simulate the front end

- added arrows to the grid lines
- added width and color to the grid lines to show power
- keep track of power on the grid lines via the graph

### CODE REVIEW

**Review by vitashubin — CHANGES_REQUESTED** [review]
(1) I'm not seeing the results page in the simulation if I pick dates longer than 1 week ago.
(2) The date selection still displays a calendar, but we can only get data based on the last week. I think it's really important to change this UI before next week's demo to reflect the actual functionality

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css [hunk 1]
```diff
@@ -3,17 +3,29 @@
...
 }
 
+.simboard-container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  flex-grow: 1;
+}
+
+
```
Comment (vitashubin) [inline biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css]: remove extra lines

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css [hunk 2]
```diff
@@ -3,17 +3,29 @@
...
   background-color: #f2f2f2;
   border-radius: 0.3125rem;
   font-family: sans-serif;
-  margin: auto;
+  margin: 2rem;
   padding: 1.25rem;
   text-align: center;
-  width: 56.25rem;
+  min-width: 56.25rem;
 }
```
Comment (vitashubin) [inline biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css:37]: looks like the simulation chart + table right now do not fit on the grey background. Please make sure your components fit inside :)

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx [hunk 3]
```diff
@@ -51,64 +53,97 @@ export const SimulatePage = () => {
       return {
         fromItem: edge.v,
         toItem: edge.w,
+        // TODO this is not a correct value
```
Comment (vitashubin) [inline biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx]: the comment behind the TODO isn't very actionable. What do we do in order to correct it?

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx [hunk 4]
```diff
@@ -51,64 +53,97 @@ export const SimulatePage = () => {
...
+        powerThroughLinesKiloWatts: edge.value.power,
       };
     });
   };
 
+  const playSimulation = () => {
```
Comment (vitashubin) [inline biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx:70]: Per UX principles, every action should be immediately un-doable by users. The "reset" button is helpful, but once they click "play", we have to give users a way to "stop" or "pause" the simulation as well.

File: biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state.spec.ts [hunk 5]
```diff
@@ -88,15 +100,15 @@ describe('classes', () => {
     expect(state.getAllVertices().length).toEqual(expectedGraph.nodes().length);
 
     // The expected object return from the shortest distances must be the same
-    expect(Object.keys(actualShortestDistances).length).toEqual(Object.keys(expectedShortestdistances).length);
+    expect(Object.keys(actualShortestDistances).length).toEqual(
+      Object.keys(expectedShortestdistances).length
+    );
     expect(actualShortestDistances).toEqual(expectedShortestdistances);
```
Comment (vitashubin) [inline biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state.spec.ts:102]: I think this last expect statement is enough, and you don't have to explicitly compare lengths before

**Review by vitashubin — CHANGES_REQUESTED** [review]

File: biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts [hunk 6]
```diff
@@ -179,24 +207,33 @@ export class Biogrid implements Grid {
...
         }
-        energyUser.startCharging(energyUserReq);
+        (energyUser as BioBattery).startCharging(energyUserReq);
         clonedGraph.setNode(energyUser.gridItemName, energyUser);
       }
+      powerEdges.push({
+        v: supplyingGridItem.gridItemName,
+        w: energyUser.gridItemName,
+        // convert kilojoules into kilowatts
+        power: energyUserReq / (TIME.DISCRETE_UNIT_HOURS * 60 * 60),
```
Comment (RolandNaijuka) [inline biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts]: Energy from the battery is already in kilowatts, there is no need to change it.
If this is for solar panels, you can change it before `lines 215 - 216`. Or you can change it entirely in `Solar Panel class`

**Review by RolandNaijuka — CHANGES_REQUESTED** [review]
Just that one change and it should be set.

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx (hunk 3 above)
Comment (Lev-Stambler) [reply biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx]: Woops, that was a note to myself. It is correct now

File: biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts (hunk 6 above)
Comment (Lev-Stambler) [reply biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts]: Yeah this is fixed in #85, energy is in kilowatthour and is then converted to kilowatt by dividing by TIME.DISCRETE_UNIT_HOURS (which is currently 1)

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css (hunk 2 above)
Comment (Lev-Stambler) [reply biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css:37]: Added to issue #86

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx (hunk 4 above)
Comment (Lev-Stambler) [reply biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx:70]: Added to issue #86

**Comment by Lev-Stambler** [comment]
> (1) I'm not seeing the results page in the simulation if I pick dates longer than 1 week ago.
> (2) The date selection still displays a calendar, but we can only get data based on the last week. I think it's really important to change this UI before next week's demo to reflect the actual functionality

Added to issue #86

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css (hunk 2 above)
Comment (Lev-Stambler) [reply biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css:37]: Fixed with overflow-y
Screenshot_2020-07-30_152641

Comment (vitashubin) [reply biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css:37]: looks great, thanks! would prefer to have a bit more space between the table and the image, can you file a bug to fix that in the future? Thanks!

Comment (vitashubin) [reply biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.css:37]: I also don't see the "Stop" button in the demo link - please make sure to test it in the UI and make sure it works as expected :)

**Review by vitashubin — APPROVED** [review]

**Pull request merged**